src/library/module.cpp

```diff
@@ -49,17 +49,26 @@ namespace lean {
 struct olean_header {
     // 5 bytes: magic number
     char marker[5] = {'o', 'l', 'e', 'a', 'n'};
-    // 1 byte: version, currently always `1`
-    uint8_t version = 1;
-    // 42 bytes: build githash, padded with `\0` to the right
-    char githash[42];
+    // 1 byte: version, incremented on structural changes to header
+    uint8_t version = 2;
+    // 1 byte flags:
+    // - whether persisted bignums use GMP or Lean-native encoding
+    bool:1 uses_gmp =
+#ifdef LEAN_USE_GMP
+        true;
+#else
+        false;
+#endif
+    bool:7 unused = 0;
+    // 41 bytes: build githash, padded with `\0` to the right
```

Comment:
This is the hex-encoded git hash rather than the raw 20-byte SHA, right?

Reply:
Yes